src/realm/cluster_tree.hpp

```diff
@@ -172,19 +198,29 @@ class ClusterTree {
     friend class ClusterNodeInner;
 
     Allocator& m_alloc;
+    Table* m_owner;
```

Comment:
We use this ptr everywhere, and we initialize it in the constructor of the cluster tree. Probably, asserting that the owner ptr passed in the ctor is not NULL could prevent to catch possible errors while using the `ClusterTree`

Reply:
That would be ideal. Unfortunately we still use a ClusterTree in a situation where the owner is a nullptr. It is in the Dictionary::migrate() function.

Reply:
OK, but does this mean that we could potentially crash? Because we use it unprotected in some places..